### ISSUE TITLE
Logic for numerical gradientcheck

### ISSUE THREAD

**Issue by suhaspillai** [sidd4698/deep-learning#2]
Hi Siddesh,

Refer the gradient check code. Also, check my code on data_load ...try to use object of numjava class for matrix computation rather than directly using an array....because I have written most of the code and  matrix functions using numjava object----> which has a matrix.

so the call to the matrix is
numjava n= new ....
n.finalmatrix------> which will give you the matrix for computation.

So, why we need this is because we want to verify the whether our back propagation is correct or not.

In Short, what we calculate using partial derivatives is analytical gradient and numerical gradient (the code which you will write ) helps to verify whether our analytical gradient calculation is correct or not.

eval_numerical_gradient()
eval_numerical_gradient_array()
write only for these two functions.

Regards,
Suhas.Pillai
